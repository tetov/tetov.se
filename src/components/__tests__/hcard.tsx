import React from "react";
import renderer from "react-test-renderer";
import type { PureHCardProp } from "src/components/hcard";
import HCard, { PureHCard } from "src/components/hcard";

describe("HCard", () => {
  const contactDataList: PureHCardProp[] = [
    {
      hcard: "p-name",
      text: "Anton Tetov Johansson",
      rel: "me",
      label: "name",
      username: "",
    },
    {
      hcard: "p-gender-identity",
      text: "(they/them)",
      url: "https://en.wikipedia.org/wiki/Non-binary_gender",
      rel: "external",
      label: "gender-identity",
      username: "",
    },
    {
      hcard: "u-email",
      text: "email",
      url: "mailto:anton@tetov.se",
      rel: "me",
      label: "email",
      username: "",
    },
    {
      hcard: "u-key",
      text: "openpgp public key",
      url: "/key.txt",
      rel: "pgpkey authn key me",
      label: "openpgp",
      username: "",
    },
  ];
  it("PureHCard renders correctly with hand converted data", () => {
    const tree = renderer
      .create(<PureHCard contactDetails={contactDataList} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  const contactDataYamlNode: Queries.ContactDataYaml = {
    id: "uuid",
    children: [],
    parent: null,
    internal: {
      contentDigest: "",
      owner: "",
      type: "",
      content: null,
      contentFilePath: null,
      description: null,
      fieldOwners: null,
      ignoreType: null,
      mediaType: null,
    },
    contactDataList,
  };
  it("HCard renders correctly with hand converted data", () => {
    const tree = renderer
      .create(<HCard nodes={[contactDataYamlNode]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
