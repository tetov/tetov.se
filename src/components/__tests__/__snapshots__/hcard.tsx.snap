// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HCard HCard renders correctly with hand converted data 1`] = `
<div
  aria-hidden={true}
  className="h-card hidden"
>
  <div
    className="inline-block group"
  >
    <span
      className="p-name"
    >
      Anton Tetov Johansson
    </span>
  </div>
  <div
    className="inline-block group"
  >
    <a
      className="p-gender-identity"
      href="https://en.wikipedia.org/wiki/Non-binary_gender"
      rel="external"
    >
      (they/them)
    </a>
  </div>
  <div
    className="inline-block group"
  >
    <a
      className="u-email"
      href="mailto:anton@tetov.se"
      rel="me"
    >
      email
    </a>
  </div>
  <div
    className="inline-block group"
  >
    <a
      className="u-key"
      href="/key.txt"
      rel="pgpkey authn key me"
    >
      openpgp public key
    </a>
  </div>
</div>
`;

exports[`HCard PureHCard renders correctly with hand converted data 1`] = `
<div
  aria-hidden={true}
  className="h-card hidden"
>
  <div
    className="inline-block group"
  >
    <span
      className="p-name"
    >
      Anton Tetov Johansson
    </span>
  </div>
  <div
    className="inline-block group"
  >
    <a
      className="p-gender-identity"
      href="https://en.wikipedia.org/wiki/Non-binary_gender"
      rel="external"
    >
      (they/them)
    </a>
  </div>
  <div
    className="inline-block group"
  >
    <a
      className="u-email"
      href="mailto:anton@tetov.se"
      rel="me"
    >
      email
    </a>
  </div>
  <div
    className="inline-block group"
  >
    <a
      className="u-key"
      href="/key.txt"
      rel="pgpkey authn key me"
    >
      openpgp public key
    </a>
  </div>
</div>
`;
